src/main/kotlin/no/nav/familie/ef/sak/task/GrensesnittavstemmingTask.kt

```diff
@@ -19,23 +22,41 @@ class GrensesnittavstemmingTask(private val avstemmingService: AvstemmingService
     val logger: Logger = LoggerFactory.getLogger(this.javaClass)
 
     override fun doTask(task: Task) {
-        val fraTidspunkt = LocalDate.parse(task.payload).atStartOfDay()
-        val tilTidspunkt = task.triggerTid.toLocalDate().atStartOfDay()
+        val payload = objectMapper.readValue<GrensesnittavstemmingPayload>(task.payload)
 
-        logger.info("Gjør avstemming mot oppdrag fra $fraTidspunkt til ${task.triggerTid.toLocalDate()}")
+        val fraTidspunkt = payload.fraDato.atStartOfDay()
+        val tilTidspunkt = task.triggerTid.toLocalDate().atStartOfDay()
 
-        avstemmingService.grensesnittavstemOppdrag(fraTidspunkt, tilTidspunkt)
+        logger.info("Gjør ${task.id} ${payload.stønadstype} avstemming mot oppdrag fra $fraTidspunkt til $tilTidspunkt")
 
+        when (payload.stønadstype) {
```

Comment:
Jeg tror dette hadde blitt ryddigere med en task per stønadstype. I stedet fo rå legge til Grensesnittavstemmingpayload og when, så burde tasken fått nytt navn GrensesnittavstemmingOvergangsstønadTask og så kan vi legge til GrensesnittavstemmingBarnetilsynTask og GrensesnittavstemmingSkolepengerTask. Mindre ansvar på hver task og renere logikk i hver av dem.

Reply:
Liker den SRP tankegången, men er osikker på hva du mener med en task per stønadstype. Mener du att vi ska lage tre bønner av typen GrensesnittAvstemmingtask som tar in stønadstype som parameter. Eller ska vi lage tre forskjellige services: GrensesnittavstemmingOvergangsstønadTask, GrensesnittavstemmingBarnetilsynTask, GrensesnittavstemmingSkolepengerTask

Reply:
Hvis du mener at vi skal lage tre ulike task-klasser så tror jeg det skalerer ganske dårlig. Da vil vi få 3 for grensesnittavstemming, 3 for konsistensavstemming, 3 for iverksetting og 3 for kvitteringshåndtering. Og alle 3 i hver kategori vil ha duplikat kode. Det skalerer ikke så bra i mine øyne og blir mange ulike tasktyper å holde styr på i tillegg til alle andre tasktyper vi vil ha i ef-sak på sikt. When-logikken vi har her er lagt inn fordi vi kun støtter overgangsstønad i påvente av klassekoder for barnetilsyn og skolepenger. Da vil vi at det skal være synlig at kun overgangsstønad er implementert. På sikt kan vi sende stønadstype rett inn til avstemmingservice.